src/client/datascience/jupyter/jupyterDebugger.ts

```
@@ -88,6 +88,9 @@ export class JupyterDebugger implements IJupyterDebugger, ICellHashListener {
             //     };
             //     await this.debugService.activeDebugSession.customRequest('setExceptionBreakpoints', args);
             // }
+        } else {
+            // if we cannot connect to the server, throw so we exit out of debugging
+            throw new JupyterDebuggerNotInstalledError(localize.DataScience.environmentVariablePtvsdLogDirIsSet());
```

>environmentVariablePtvsdLogDirIsSet [](start = 76, length = 35)

This is very specific. There could be any number of reasons why ptvsd fails to launch. 

Instead you should just parse the output from parseConnectInfo and send that in a JupyterDebuggerFailedError if parseConnectInfo fails.

---

When it fails it returns undefined, there's nothing to parse. I can put the exception in connect, after connectToLocal. Would that be better?

---

Instead of returning undefined, throw an exception.

---
In reply to: [313600304](https://github.com/microsoft/vscode-python/pull/6956#discussion_r313600304) [](ancestors = 313600304)

---

Where the message for the exception is the output from the ptvsd command.

---
In reply to: [313600655](https://github.com/microsoft/vscode-python/pull/6956#discussion_r313600655) [](ancestors = 313600655,313600304)

---

I changed it to throw the exception on parseConnectInfo, but the extract also returns undefined.